src/layout/msWorkspace/horizontalPanel.js

```diff
@@ -49,12 +49,14 @@ var HorizontalPanel = GObject.registerClass(
             if (Me.msThemeManager.clockHorizontal) {
                 this.createClock();
             }
-            Me.msThemeManager.connect('panel-size-changed', () => {
-                this.tilingIcon.set_icon_size(
-                    Me.msThemeManager.getPanelSizeNotScaled() / 2
-                );
-                this.queue_relayout();
-            });
+        }
+
+        buildIcon(height) {
+            this.iconSize = height;
+            this.tilingIcon.set_icon_size(
+                Me.msThemeManager.getPanelSizeNotScaled() / 2
+            );
+            this.queue_relayout();
```

Comment:
This trigger vfunc_allocate which is the one one call this function. It's could easily lead to a nasty loop and I'm not sure it's usefull in this case

Reply:
It is the same code used here:
https://github.com/material-shell/material-shell/blob/master/src/widget/taskBar.js#L642-L650

There is no loop because it checks iconSize value.

Reply:
I know there is a check to prevent the loop but it's still bad practice for this future us and mostly because I think it's useless if you change the size in the allocate function

Reply:
Only actors visible in the stage can be updated this way. See my first comment about `panel-size-changed` signal. Even the stylesheet theme isn't reloaded correctly for the top panels that are not visible. The only other way to avoid this is to kill and reload the whole extension.